applications/GeoMechanicsApplication/custom_constitutive/linear_elastic_law.cpp

```diff
@@ -185,6 +186,15 @@ bool GeoLinearElasticLaw::GetConsiderDiagonalEntriesOnlyAndNoShear() const
     return mConsiderDiagonalEntriesOnlyAndNoShear;
 }
 
+void GeoLinearElasticLaw::CalculateCauchyGreenStrain(ConstitutiveLaw::Parameters& rValues, Vector& rStrainVector)
+{
+    Matrix F = rValues.GetDeformationGradientF();
+    if (this->WorkingSpaceDimension() == 2) F.resize(2,2, true);
+
+    Matrix E_tensor        = StressStrainUtilities::CalculateGreenLagrangeStrainTensor(F);
+    noalias(rStrainVector) = MathUtils<double>::StrainTensorToVector(E_tensor);
+}
```

Comment:
I moved this from the k0 law to here and made it more generic for 2d/3d. However, we have very similar functionality in the stress state policy (where the end result is slightly different) and since this is 2d/3d specific, I'd like to discuss how we want to handle this.